velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -284,16 +289,46 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
       if (isFileColumnNamesReadAsLowerCase()) {
         folly::toLowerAscii(childName);
       }
-      auto childRequestedType =
-          requestedType ? requestedType->asRow().findChild(childName) : nullptr;
+
+      TypePtr childRequestedType = nullptr;
+      if (requestedType) {
+        if (requestedType->isRow()) {
+          auto fileTypeIdx =
+              requestedType->asRow().getChildIdxIfExists(childName);
+          if (fileTypeIdx.has_value()) {
+            childRequestedType = requestedType->asRow().childAt(*fileTypeIdx);
+          }
+        }
+
+        if (parentRequestedType) {
+          if (parentRequestedType->isArray()) {
+            childRequestedType = parentRequestedType->asArray().elementType();
+          } else if (parentRequestedType->isMap()) {
+            auto mapType = parentRequestedType->asMap();
+            auto isMapKey = schemaElement.__isset.repetition_type &&
+                schemaElement.repetition_type ==
+                    thrift::FieldRepetitionType::REPEATED &&
+                schema[parentSchemaIdx].converted_type ==
+                    thrift::ConvertedType::MAP &&
+                i == 0;
+            if (isMapKey) {
+              childRequestedType = mapType.keyType();
+            } else {
+              childRequestedType = mapType.valueType();
+            }
+          }
+        }
+      }
+
       auto child = getParquetColumnInfo(
           maxSchemaElementIdx,
           maxRepeat,
           maxDefine,
           curSchemaIdx,
           schemaIdx,
           columnIdx,
-          childRequestedType);
+          childRequestedType ? childRequestedType : requestedType,
```

Comment:
Should we just pass `childRequestedType` in this case?  What's the purpose of passing `requestedType` here?

Reply:
Or we can move the following code out of `if(requestedType) {...}`:
```
        if (parentRequestedType) {
          if (parentRequestedType->isArray()) {
            childRequestedType = parentRequestedType->asArray().elementType();
          } else if (parentRequestedType->isMap()) {
            auto mapType = parentRequestedType->asMap();
            auto isMapKey = schemaElement.__isset.repetition_type &&
                schemaElement.repetition_type ==
                    thrift::FieldRepetitionType::REPEATED &&
                schema[parentSchemaIdx].converted_type ==
                    thrift::ConvertedType::MAP &&
                i == 0;
            if (isMapKey) {
              childRequestedType = mapType.keyType();
            } else {
              childRequestedType = mapType.valueType();
            }
          }
        }
```

Reply:
@wypb let's make this change. It is cleaner. We can combine the second block check
`if(!requestedType && parentRequestedType) {`

Reply:
Hi @majetideepak I have modified the corresponding code as suggested. Thank you for your review.